modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/ClickGoalPicker/ClickGoalPicker.es.js

```diff
@@ -99,7 +98,7 @@ function ClickGoalPicker({allowEdit = true, onSelectClickGoalTarget, target}) {
 	}
 
 	const scrollIntoView = (event) => {
-		const target = document.querySelector(state.selectedTarget);
+		const target = document.querySelector(`#${state.selectedTarget}`);
```

Comment:
You never want to use `querySelector` with a dynamically constructed string starting with `#` like this. Instead, use `document.getElementById(state.selectedTarget)` (AKA "Use The Right Tool For The Job™"; `getElementById()` is probably faster, and you avoid the cost and ugliness of the string interpolation).

See [discussion in liferay-frontend-guidelines repo](https://github.com/liferay/liferay-frontend-guidelines/issues/19). We never got around to documenting it, but it's reflected there in the issue.

Reply:
You are totally true. I'm going to change it 👍